pkg/reconciler/ingress/resources/gateway.go

```diff
@@ -83,9 +89,77 @@ func SortServers(servers []v1alpha3.Server) []v1alpha3.Server {
 	return servers
 }
 
-// MakeServers creates the expected Gateway `Servers` based on the given
-// ClusterIngress.
-func MakeServers(ia v1alpha1.IngressAccessor, gatewayServiceNamespace string, originSecrets map[string]*corev1.Secret) ([]v1alpha3.Server, error) {
+// MakeIngressGateways creates Gateways for a given IngressAccessor.
+func MakeIngressGateways(ctx context.Context, ia v1alpha1.IngressAccessor, originSecrets map[string]*corev1.Secret, svcLister corev1listers.ServiceLister) ([]*v1alpha3.Gateway, error) {
+	gatewayServices, err := getGatewayServices(ctx, svcLister)
+	if err != nil {
+		return nil, err
+	}
+	gateways := []*v1alpha3.Gateway{}
```

Comment:
`make([]*v1alpha3.Gateway{}, 0, len(gatewayServices)`
or `make([]*v1alpha3.Gateway{}, len(gatewayServices)` and use index setter.

Reply:
Done